cmd/olympus/main.go

```diff
@@ -1,48 +1,87 @@
 package main
 
 import (
-	"fmt"
 	"log"
 
+	"github.com/garyburd/redigo/redis"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/worker"
+	"github.com/gobuffalo/gocraft-work-adapter"
 	"github.com/gomods/athens/cmd/olympus/actions"
+	"github.com/gomods/athens/pkg/config/env"
+	"github.com/gomods/athens/pkg/eventlog"
 	"github.com/gomods/athens/pkg/storage"
 )
 
 func main() {
-	app := actions.App()
+	app := setupApp()
 
-	s, err := getLocalStorage()
-	if err != nil {
+	if err := app.Serve(); err != nil {
 		log.Fatal(err)
 	}
+}
 
-	e, err := actions.GetEventLog()
-	if err != nil {
-		log.Fatal(err)
+func setupApp() *buffalo.App {
+	w := getWorker()
+	storage := getStorage()
+	eLog := getEventLog()
+	cacheMissesLog := getCacheMissesEventLog()
+	config := actions.AppConfig{
+		Storage:        storage,
+		EventLog:       eLog,
+		CacheMissesLog: cacheMissesLog,
+		Worker:         w,
 	}
+	app := actions.App(&config)
 
-	w := app.Worker
-	if err := w.Register(actions.DownloadHandlerName, actions.GetPackageDownloaderJob(s, e, w)); err != nil {
+	if err := w.Register(actions.DownloadHandlerName, actions.GetPackageDownloaderJob(storage, eLog, app.Worker)); err != nil {
 		log.Fatal(err)
 	}
-	if err := w.Register(actions.PushNotificationHandlerName, actions.GetProcessPushNotificationJob(s, e, w)); err != nil {
+	if err := w.Register(actions.PushNotificationHandlerName, actions.GetProcessPushNotificationJob(storage, eLog)); err != nil {
 		log.Fatal(err)
 	}
+	return app
+}
 
-	if err := app.Serve(); err != nil {
-		log.Fatal(err)
+func getCacheMissesEventLog() eventlog.Appender {
+	cacheMissesLog, err := actions.NewCacheMissesLog()
+	if err != nil {
+		log.Fatalf("error creating cachemisses log (%s)", err)
```

Comment:
do you mind adding an error return value here, so that the caller can decide what to do on error?

Reply:
if this is the case, we can specify custom err types, so decisions are easier